game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/CheckGeneralBattleEnd.java

```diff
@@ -54,12 +54,15 @@ public Order getOrder() {
   public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
     if (hasSideLost(OFFENSE)) {
       battleActions.endBattle(IBattle.WhoWon.DEFENDER, bridge);
+      stack.clear();
```

Comment:
This class seems misleading in name as the 'general' battle phase is just one phase and perhaps might not be the last phase of combat (it just so happens to be for now). Would this be clarified if this class were simply renamed to `CheckBattleEnd`?


Side-note - At some point too I think we should think about having a proper `Battle` object with a defined API. I'm not suggesting we go back to having a `MustFightBattle` that has everything in it again, but we went from that to its components pieces being broken out, I think re-combining them for a class to use those components pieces could be appropriate. In this way we'd have a lot of package protected strategy objects with very few entry points, the interactions would be mediated by more or less a single object. In other words, instead of many parts of the system having access to battle state and way to change battle state, the ways battle state would be changed would be mediated by more or less a single object.

Reply:
Renaming isn't possible until we break save compatibility.

I see `BattleState` as the object with a defined API.  All of the component pieces are the strategy objects that interact with the `BattleState` and perform the logic needed.  I hope to start converting the other battle classes (`NonFightingBattle`, `AirBattle`, `DependentBattle`, `StrategicBombingRaidBattle`, `FinishedBattle`) to use `BattleState` and these strategy objects.

As for combining them all in one class, it isn't possible because each of the pieces have to be a unique `IExecutable`. And defining all of the `IExecutables` in one class makes it very hard to make changes since the order of the definition of the `IExecutable` is now important for save compatibility.

Reply:
One class is not necessarily the objective, it's more to reduce the 'fan-in'. Let me know if that makes more sense.